Features/Swap/Sources/ViewModels/SwapSceneViewModel.swift

```diff
@@ -382,25 +384,26 @@ extension SwapSceneViewModel {
         )
     }
 
-    public func swap() {
+    private func swap() {
         guard let fromAsset = fromAsset,
               let toAsset = toAsset,
               let quote = selectedSwapQuote
         else {
             return
         }
-        swapTransferData = SwapTransferDataFactory.swap(
-            fromAsset: fromAsset.asset,
-            toAsset: toAsset.asset,
-            quote: quote,
-            quoteData: nil
+        onSwap?(
+            SwapTransferDataFactory.swap(
```

Comment:
```suggestion
            SwapTransferDataFactory.swap(
```

because it's a factory, it need initializer from `SwapTransferDataFactory.from(quote:)`, quote has access to from/to asset